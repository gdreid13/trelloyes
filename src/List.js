import React from 'react';
import './list.css';
import Card from './Card'
import STORE from './store';

function List(props){
  const cards = props.cardIds.map(cardId =>
    <Card 
        title = {props.allCards[cardId].title}
        content = {props.allCards[cardId].content}
        key = {cardId}
    />
  );
  return (
  <section className="List">
    <header className="List-header">{props.header}</header>
    <div className="List-cards">{cards}</div>
  </section>
  )
};

List.defaultProps = {
  header : "header",
  cardIds : ['a', 'b', 'c'],
  allCards : STORE.allCards,
}

export default List;