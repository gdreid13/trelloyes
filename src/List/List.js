import React from 'react';
import list from '../List/list.css'
import Card from '../Card/Card'

function List(props){
  <section className="List">
    <header className="List-header">{props.header}</header>
    <div className="List-cards">{props.cards}</div>
  </section>
}

export default List 