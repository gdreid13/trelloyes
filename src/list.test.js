import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';
import STORE from './store';

describe('List component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List />, div);
        ReactDOM.unmountComponentAtNode(div);
    }); 
    
    it ('snapshot of list', () => {
        const snap = renderer.create(<List />).toJSON();
        expect(snap).toMatchSnapshot();
    });
});