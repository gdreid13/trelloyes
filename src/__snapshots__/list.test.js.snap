// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component snapshot of list 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    header
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Third card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
